### PR TITLE
sdl_image: don't use *_DYNAMIC options

### LINKED ISSUE #15440: sdl_image/2.0.5: PNG_LOAD_DYNAMIC should be string
### Description

Now sdl_image recipe defines `LOAD_PNG_DYNAMIC` as True or False, which leads to crashes in this piece of code due to if this defined as boolean 
```c        
lib.handle = SDL_LoadObject(LOAD_PNG_DYNAMIC);
```
(as a package, with_libpng is set). This should be set as library name instead (like libpng.so.16 for Linux).

### Package and Environment Details

* Package Name/Version: **sdl_image/2.0.5**
* Operating System+version: **ArchLinux**
* Compiler+version: **GCC 12 (MinGW-Cross)**
* Conan version: **conan 1.57.0**
* Python version: **Python 3.10**

### Conan profile

toolchain=/usr/i686-w64-mingw32
target_host=i686-w64-mingw32

[settings]
os=Windows
arch=x86
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=10
build_type=Release
compiler.cppstd=11

[conf]
tools.cmake.cmaketoolchain:generator = Ninja
tools.build:compiler_executables = {"c": "i686-w64-mingw32-gcc", "cpp": "i686-w64-mingw32-g++"}
tools.build:cflags = ["-msse2"]
tools.build:cxxflags = ["-msse2"]
tools.build:sysroot = $toolchain

[env]
CONAN_CROSS_COMPILE=$target_host-
CONAN_SYSTEM_LIBRARY_LOCATION=/usr/lib/gcc/i686-w64-mingw32/10-posix/
CHOST=$target_host
AR=$target_host-ar
AS=$target_host-as
CC=$target_host-gcc
CXX=$target_host-g++
RANLIB=$target_host-ranlib
STRIP=$target_host-strip
RC=$target_host-windres

### Steps to reproduce

Try to compile and run something which use IMG_Load_RW with PNG image, for example, Battle for Wesnoth or VCMI.

### Logs

<details><summary>Click to expand log</summary>

```
Thread 2 "01e4" received signal SIGSEGV, Segmentation fault.
```

</details>

### PR DESCRIPTION
**Pull request by kambala-decapitator**
Specify library name and version:  **sdl_image/2.0.5**

`*_DYNAMIC` options are meant for `dlopen` and similar, but setting them also requires shared library name to be set to `-DLOAD_*_DYNAMIC` macro which is not the case. Current Conan's CML links all requested libs explicitly anyway, so these dynamic options are basically useless.

fixes #15440

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
There was this PR https://github.com/conan-io/conan-center-index/pull/14493, but didn't have the time to address comments of @madebr

**Comment by kambala-decapitator** [comment]
@SpaceIm do you think this PR would be fine to go in as a short-term solution after #16168 is merged? This fix is needed only for v2.0.5 anyway.

**Comment by prince-chrismc** [comment]
Is this still needed? the other PR was merged

**Comment by kambala-decapitator** [comment]
> Is this still needed? the other PR was merged

which PR do you mean? Don't see any fresh merges: https://github.com/conan-io/conan-center-index/pulls?q=is%3Apr+is%3Aclosed+sdl_image

**Comment by prince-chrismc** [comment]
> you think this PR would be fine to go in as a short-term solution after https://github.com/conan-io/conan-center-index/pull/16168 is merged? This fix is needed only for v2.0.5 anyway

**Comment by kambala-decapitator** [comment]
> > you think this PR would be fine to go in as a short-term solution after https://github.com/conan-io/conan-center-index/pull/16168 is merged? This fix is needed only for v2.0.5 anyway

Ah thanks. I mentioned it because my PR failed due to dependency conflict and that PR bumped them, so I was waiting for its merge.

File: recipes/sdl_image/all/conanfile.py
Comment (prince-chrismc) [inline recipes/sdl_image/all/conanfile.py:122]: +1 it should be removed given the converstation

**Review by prince-chrismc — APPROVED** [review]

File: recipes/sdl_image/all/conanfile.py
Comment (SSE4) [reply recipes/sdl_image/all/conanfile.py:122]: so there should be more options, like you can link statically, link dynamically or use `dlopen`/`LoadLibrary` to locate a specific image plug-in.

**Review by SSE4 — APPROVED** [review]

File: recipes/sdl_image/all/conanfile.py
Comment (SpaceIm) [reply recipes/sdl_image/all/conanfile.py:122]: there was https://github.com/conan-io/conan-center-index/pull/14493

**Pull request merged**
